lib/src/builder/generation/declaration_parsing.dart

```diff
@@ -282,6 +282,29 @@ class ParsedDeclarations {
       );
     }
 
+    // Ensure that Component2 declarations do not use legacy lifecycle methods.
+
+    if (declarationMap[key_component2].isNotEmpty) {
+      if (declarationMap[key_component2].first is ClassDeclaration) {
+        ClassDeclaration componentObject = declarationMap[key_component2].first;
+        List<String> legacyLifecycleEvents =
+            ['componentWillReceiveProps', 'componentWillMount', 'componentWillUpdate'];
```

Comment:
I just noticed that `componentWillMount` is not deprecated.  I'm assuming we should do that in react-dart 5.0.0-wip, and recommend the use of `init` instead?

Reply:
It's not deprecated in react-dart 5.0.0-wip, but it is in 5.1.0-wip. Since 5.0.0 doesn't have the new lifecycle methods, would the deprecation messages be just to give heads up not to rely on any of the legacy methods?

Reply:
Hmmm... thats weird... when I pulled this branch down yesterday I didn't see any deprecation messages for `componentWillMount`...  Let me double-check.

Reply:
nothing to see here... just had some cached deps that were out of date. all good!